src/components/WalletList.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import './WalletList.css';
+
+type WalletEntryParams = {
+  name: string;
+  uuid: string;
+  rdns: string;
+  icon: string;
+  extensionId?: string;
+};
+
+export type WalletMapEntry = {
+  params: WalletEntryParams;
+  eventName: string;
+};
+
+type WalletListProps = {
+  wallets: Record<string, WalletMapEntry>;
+  handleClick: (extensionId: string) => void;
+};
+
+const WalletList: React.FC<WalletListProps> = ({ wallets, handleClick }) => {
+  const walletEntries = Object.entries(wallets);
+
+  if (!walletEntries.length) {
+    return null;
+  }
+
+  return (
+    <div className="wallet-list">
+      {walletEntries.map(([key, wallet]) => (
+        <div key={key} className="wallet-card">
+          <img
+            src={wallet.params.icon}
+            alt={`${wallet.params.name} icon`}
+            className="wallet-icon"
+          />
+          <div className="wallet-info">
+            <h3 className="wallet-name">{wallet.params.name}</h3>
+            <p className="wallet-uuid">UUID: {wallet.params.uuid}</p>
+            <p className="wallet-rdns">RDNS: {wallet.params.rdns}</p>
+            {wallet.params.extensionId && (
+              <p className="wallet-extension-id">
+                Extension ID: {wallet.params.extensionId}
+              </p>
+            )}
+          </div>
+          <button
+            disabled={Boolean(!wallet.params.extensionId)}
```

Comment:
maybe just `!!` instead? I know some people would disagree with me. Up to you, but at minimum i think it's a little odd to have the `!` negate inside the `Boolean()`

Reply:
<img width="484" alt="Screenshot 2024-12-12 at 21 47 39" src="https://github.com/user-attachments/assets/0419443d-a049-4ca5-b33c-9f9b398dd4a3" />
we have eslint rule for this it seems 😱 So have to resort to wrapping in `Boolean` and negating it